lib/sycamore/sycamore/llms/openai.py

```diff
@@ -350,41 +365,15 @@ def is_chat_mode(self):
         return self.model.is_chat
 
     def _get_cache_key(self, prompt_kwargs: dict, llm_kwargs: Optional[dict] = None) -> str:
+        """Override _get_cache_key to include response_format param."""
         assert self._cache
-        combined = {"prompt_kwargs": prompt_kwargs, "llm_kwargs": llm_kwargs, "model_name": self.model.name}
-        data = pickle.dumps(combined)
-        return self._cache.get_hash_context(data).hexdigest()
-
-    def _cache_get(self, prompt_kwargs: dict, llm_kwargs: Optional[dict] = None):
-        if (llm_kwargs or {}).get("temperature", 0) != 0 or not self._cache:
-            # Never cache when temperature setting is nonzero.
-            return (None, None)
-
+        llm_kwargs = llm_kwargs.copy() if llm_kwargs else {}
```

Comment:
Bug/bug reveal: The previous code was working because we were modifying llm_kwargs and serializing the `response_type` value in _get_cache_key. Now that we're copying, the llm_kwargs still has the unserialized value and that fails to pickle. If you drop this line it should work but you probably want a better solution

Reply:
Fixed.